test/stream_test/irods_test_read_object.cpp

```diff
@@ -0,0 +1,118 @@
+#include <irods/irods_at_scope_exit.hpp>
+#include <irods/client_connection.hpp>
+#include <irods/dataObjClose.h>
+#include <irods/dataObjOpen.h>
+#include <irods/dataObjRead.h>
+#include <irods/rodsClient.h>
+
+#include <fmt/format.h>
+
+#include <boost/program_options.hpp>
+
+#include <iostream>
+#include <string>
+
+// This program opens the specified iRODS data object, reads the specified number of bytes, and prints it to stdout.
+//
+// Assumes an authenticated iRODS client environment is available in the executing user's ~/.irods directory.
+// The program prints the contents to stdout. Would not recommend using this on anything other than text files.
+//
+// NOTE: This program is only intended for testing. Do not use in production.
```

Comment:
Could put this in the help text for when we rediscover this tool 6 months from now.

Reply:
Done